include/bout/region.hxx

```diff
@@ -643,8 +655,7 @@ public:
     return *this; // To allow command chaining
   };
 
-  /// Returns a new region including only indices contained in both
-  /// this region and the other.
+  /// Returns a modified region including only indices that are also in the region.
```

Comment:
I think @johnomotani encountered something about this function before. This modifies the object in-place as well as returning a new instance. Maybe we want to make a bigger change, like make this `const` and not modify in-place?

Reply:
I added a test, because I didn't believe the comment. This function is used to create a new instance, and thus this function is needed, at least internally. It is not used by anything else, thus we could refactor it and move it to the free function that creates a new object.

Reply:
:+1:  Good idea! Maybe rename `getIntersection` -> `intersection` for the free function?

Reply:
Done :+1: